cluster/src/main/scala/org/apache/spark/memory/SnappyUnifiedMemoryManager.scala

```diff
@@ -360,6 +361,15 @@ class SnappyUnifiedMemoryManager private[memory](
           return
         }
 
+        // Stop execution pool to grow beyond eviction percentage of heap.
+        if (memoryMode eq MemoryMode.ON_HEAP) {
+          if (executionPool.memoryUsed + extraMemoryNeeded > maxHeapExecutionSize) {
+            logWarning(s"MemoryManager can't allocate $numBytes bytes as it " +
+                s"would exceed maxHeapExecutionSize = $maxHeapExecutionSize")
+            return
+          }
+        }
+
```

Comment:
No, if at a time store also wants the area it would evict its own area. As execution memory is temporary after some time store can reclaim its area. But at that very moment store has to evict some of its data. It can not force spill execution structures.